catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/builder/AttributeValueNormalizerTest.java

```diff
@@ -0,0 +1,183 @@
+    assertThat(
+        normalizer.normalizeForJson(PROPERTY_NAME, NON_DATE_INPUT), is(equalTo(NON_DATE_INPUT)));
+  }
+
+  @Test
+  public void testNonDateValueToXml() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.empty());
+    assertThat(
+        normalizer.normalizeForXml(PROPERTY_NAME, NON_DATE_INPUT), is(equalTo(NON_DATE_INPUT)));
+  }
+
+  @Test
+  public void testEpochDateValueToJson() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForJson(PROPERTY_NAME, VALID_MS_SINCE_EPOCH_DATE_STRING),
+        is(equalTo(VALID_ISO_8601_DATE_STRING)));
+  }
+
+  @Test
+  public void testIsoDateValueToJson() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForJson(PROPERTY_NAME, VALID_ISO_8601_DATE_STRING),
+        is(equalTo(VALID_ISO_8601_DATE_STRING)));
+  }
+
+  @Test
+  public void testUnexpectedDateValueToJson() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForJson(PROPERTY_NAME, NON_DATE_INPUT), is(equalTo(NON_DATE_INPUT)));
+  }
+
+  @Test
+  public void testEpochDateValueToXml() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForXml(PROPERTY_NAME, VALID_MS_SINCE_EPOCH_DATE_STRING),
+        is(equalTo(VALID_MS_SINCE_EPOCH_DATE_STRING)));
+  }
+
+  @Test
+  public void testIsoDateValueToXml() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForXml(PROPERTY_NAME, VALID_ISO_8601_DATE_STRING),
+        is(equalTo(VALID_MS_SINCE_EPOCH_DATE_STRING)));
+  }
+
+  @Test
+  public void testIsoDateWithOffsetValueToXml() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    assertThat(
+        normalizer.normalizeForXml(PROPERTY_NAME, VALID_ISO_8601_DATE_STRING_WITH_OFFSET),
+        is(equalTo(VALID_MS_SINCE_EPOCH_DATE_STRING_WITH_OFFSET)));
+  }
+
+  @Test
+  public void testBadPropertyNameToJson() {
+    when(registry.lookup(eq(PROPERTY_NAME_WITH_QUOTES))).thenReturn(Optional.empty());
+    assertThat(
+        normalizer.normalizeForJson(PROPERTY_NAME_WITH_QUOTES, VALID_ISO_8601_DATE_STRING),
+        is(equalTo(VALID_ISO_8601_DATE_STRING)));
+  }
+
+  @Test(expected = FilterProcessingException.class)
+  public void testBadPropertyNameToXml() {
+    normalizer.normalizeForXml(PROPERTY_NAME_WITH_QUOTES, VALID_ISO_8601_DATE_STRING);
+  }
+
+  @Test(expected = FilterProcessingException.class)
+  public void testUnexpectedDateValueToXml() {
+    when(registry.lookup(eq(PROPERTY_NAME))).thenReturn(Optional.of(DATE_DESCRIPTOR));
+    normalizer.normalizeForXml(PROPERTY_NAME, NON_DATE_INPUT);
+  }
+
+  private static AttributeDescriptor dateDescriptor() {
```

Comment:
✏️ might want to prefix with `create` or `generate`

Reply:
Fixed.